sips/sip-022/sip-022-emergency-pox-fix.md

```diff
@@ -0,0 +1,284 @@
+
+
+Furthermore, if this bug is triggered enough times, the Stacks network will crash.  This is
+because of a runtime assertion in the PoX reward set calculation logic that
+verifies that the total locked STX does not exceed the total liquid STX.  This
+would no longer be true due to this bug.  The offending assertion is detailed
+below:
+
+```rust
+pub fn get_reward_threshold_and_participation(
+    pox_settings: &PoxConstants,
+    addresses: &[RawRewardSetEntry],
+    liquid_ustx: u128,
+) -> (u128, u128) {
+    let participation = addresses
+        .iter()
+        .fold(0, |agg, entry| agg + entry.amount_stacked);
+
+    assert!(
+        participation <= liquid_ustx,
+        "CORRUPTION: More stacking participation than liquid STX"
+    );
+```
+
+The `RawRewardSetEntry` data are pulled directly from the
+`reward-cycle-pox-address-list` data map, and the `.amount_stacked` field is
+equal to the `total-ustx` field that was erroneously set in `stack-increase`.
+
+Considering these consequences with respect to the [draft blockchain catastrophic
+failure and recovery guidelines](https://github.com/stacksgov/sips/pull/10),
+this bug would be categorized as requiring an _immediate_ hard fork to rectify.
+The network has not yet crashed, but it is in imminent danger of crashing and
+there is insufficient time to execute a Stacker-based SIP vote as has been customary for
+past hard forks.  This SIP instead proposes that this hard fork activate at the
+start of the next reward cycle (#58) at the time of this writing.
+
+There is less than one reward cycle remaining to fix this
+problem, and yet a Stacker vote would require at least one complete reward cycle
+to carry out vote (not accounting for any additional time required for
+sending out adequate public communications and tabulating the vote afterwards).
+A subsequent hard fork to re-enable PoX would have the effect of restoring
+Stacker voting.
+
+# Specification
+
+Given the lack of time to conduct a Stacker vote to activate this SIP, the
+proposed fix in this SIP is as parsimonious and discreet as possible.  It will
+execute as a set of **two hard forks**.
+
+The first hard fork, which activates at the start of reward cycle #58 will disable PoX.
+This hard fork would do the following:
+
+* Prevent all stacking functions from being called in `pox-2`.  The `pox-2`
+  contract would be considered defunct, like the `pox` contract is now.
+
+* Set the minimum required block-commit memo bits to `0x07`.  All block-commits
+  after the Bitcoin block activation height must have a memo value of at least
+`0x07`.  This ensures that miners that do not upgrade from Stacks 2.1 will not
+be able to mine in Stacks 2.2.
+
+* Set the mainnet peer network version bits to `0x18000007`.  This ensures that follower
+  nodes that do not upgrade to Stacks 2.2 will not be able to talk to Stacks
+2.2 nodes.
+
+The second hard fork will deactivate `pox-2` and instantiate a new PoX
```

Comment:
I think the wording is correct. 
The first hard fork will *disable* pox-2, but the second hard fork will deactivate it.

Reply:
@friedger is correct.  The first hard-fork will declare `pox-2` to be defunct, just like `pox-1` is.  The second hard-fork instantiates `pox-3`.  We're doing two hard forks because we don't have time to do one.